dcpy/lifecycle/package/assemble.py

```diff
@@ -0,0 +1,40 @@
+from pathlib import Path
+import typer
+import tempfile
+import urllib.request
+import zipfile
+import shutil
+
+import dcpy.models.product.dataset.metadata as models
+from dcpy.utils.logging import logger
+from dcpy.lifecycle import WORKING_DIRECTORIES
+
+ASSEMBLY_DIR = WORKING_DIRECTORIES.packaging / ".assembly"
+
+
+def from_bytes(
+    product_metadata: models.Metadata, product: str, version: str, dataset: str
+):
+    out_dir = ASSEMBLY_DIR / product / version / dataset
+    out_dir.mkdir(exist_ok=True, parents=True)
+
+    for bytes_dest in [d for d in product_metadata.destinations if d.type == "bytes"]:
+        package_files = product_metadata.package._files_by_name()
+
+        for dest_file in bytes_dest.files:
+            f = package_files[dest_file.name]
+            file_name: str = f.filename if issubclass(type(f), models.File) else f  # type: ignore
```

Comment:
nit - you can just use `isinstance` here I believe. And for line 31, I think `isinstance` is slightly preferred over checking the returned value of `type(f)`